devito/propagator.py

```diff
@@ -254,12 +255,33 @@ def convert_equality_to_cgen(self, equality):
             return equality
         else:
             s_lhs = ccode(self.time_substitutions(equality.lhs).xreplace(self._var_map))
-            s_rhs = ccode(self.time_substitutions(equality.rhs).xreplace(self._var_map))
+            s_rhs = self.time_substitutions(equality.rhs).xreplace(self._var_map)
+
+            self.sub_stencils.append(s_rhs)  # appending substituted stencil,which is used to determine alignment pragma
+
+            s_rhs = ccode(s_rhs)
             if self.dtype is np.float32:
                 s_rhs = str(s_rhs).replace("pow", "powf")
                 s_rhs = str(s_rhs).replace("fabs", "fabsf")
             return cgen.Assign(s_lhs, s_rhs)
 
+    def set_aligned_pragma(self, stencils, factorized, loop_counters, time_steppers):
+        """
+        Sets the alignment for the pragma.
+        :param stencils: List of stencils.
+        :param factorized:  dict of factorized elements
+        :param loop_counters: list of loop counter symbols
+        :param time_steppers: list of time stepper symbols
+        """
+
+        array_names = set()
+        for item in flatten([stencil.free_symbols for stencil in stencils]):
+            if str(item) not in factorized and item not in loop_counters + time_steppers:
+                array_names.add(item)
+
+        pragma = "omp simd aligned(%s:64)" % ", ".join([str(i) for i in array_names])
+        self.compiler.pragma_aligned = cgen.Pragma(pragma)
```

Comment:
I think we should re-create this pragma per stencil, since it looks like it depends on the free variables in the stencil. I don't think the Propagator should remote set options on the compiler; instead it should get all platform and toolchain-specific information from the Compiler, like the base string of the pragma and the alignment size maybe.